src/src/context/TodosContext.tsx

```diff
@@ -0,0 +1,209 @@
+import React, { createContext, useEffect, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { Statuses } from '../types/Statuses';
+import { getTodos } from '../api/todos';
+import { USER_ID } from '../constants/USER_ID';
+import { Errors } from '../types/Errors';
+import * as TodoService from '../api/todos';
+
+type PropsTodosContext = {
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+  filter: Statuses;
+  setFilter: (value: Statuses) => void;
+  errorMessage: Errors,
+  setErrorMessage: (value: Errors) => void,
+  tempTodo: Todo | null,
+  setTempTodo: (tempTodo: Todo | null) => void,
+  title: string,
+  setTitle: (title: string) => void,
+  chosenTodoId: number[];
+  setChosenTodoId: (id: number[]) => void;
+  addTodo: (todo: Todo) => void,
+  deleteTodo: (todoId: number) => void,
+  deleteCompletedTodos: () => void;
+  upgradeTodo: (todo: Todo) => void,
+  handleToggleAll: () => void,
+};
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodosContext = createContext<PropsTodosContext>({
+  todos: [],
+  setTodos: () => { },
+  filter: Statuses.All,
+  setFilter: () => {},
+  errorMessage: Errors.NoErrors,
+  setErrorMessage: () => {},
+  tempTodo: null,
+  setTempTodo: () => {},
+  title: '',
+  setTitle: () => {},
+  chosenTodoId: [],
+  setChosenTodoId: () => {},
```

Comment:
```suggestion
  chosenTodoIds: [],
  setChosenTodoIds: () => {},
```
Plural for arrays - it's a bit more clear